### PR TITLE
Feature/ssl inspection module

### PR DESCRIPTION
**Pull request by atran-inverse**
# Description
New portal module for SSL Inspection. 
Direct users automatically on captive portal to correct page with instructions on installing certificate.

# Impacts
New portal module. Could be used in a chain.

# Delete branch after merge
YES

# NEWS file entries
## New Features
* SSL Inspection Portal Module

### CODE REVIEW

**Comment by extrafu** [comment]
@cgx Please review the UI side and @julsemaan the backend side.

**Comment by julsemaan** [comment]
I always liked to review backends

File: html/captive-portal/content/sslinspection.js
Comment (cgx) [inline html/captive-portal/content/sslinspection.js:22]: Replace the content of this file by the JavaScript you defined in `html/captive-portal/templates/ssl_inspection/common.html`. Your `testFailed` function could be directly integrated in the failure callback since it is called only at this place.

File: html/captive-portal/templates/ssl_inspection.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection.html]: String must be localized.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: Replace `Unsupported` by `Other` and localize the string.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: The `nav` block should be moved to its own file and inserted using an `INCLUDE` directive. Repeat for all files.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: This seems specific to a client.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: I would simplify this sentence to *Download the certificate to your device.* Repeat for all files.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: Pad the Template Toolkit variable with spaces (`[% ssl_path %]`).

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: The *alt* string must be localized.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: I propose the following text:

> `You will be prompted to enter a name for the certificate. Enter a meaningful name for this network and click <i>OK</i>.`

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: Remove the surrounding parentheses.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: Wrap this note as a `o-media` *warning* object.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: Localize the string of the *alt* attribute. Repeat for all files.

File: html/captive-portal/templates/ssl_inspection/chrome.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/chrome.html]: Remove the parenthesis and wrap the note around a warning `o-media` object.

File: html/captive-portal/templates/ssl_inspection/ios.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/ios.html]: This block seems specific to a client.

File: html/captive-portal/templates/ssl_inspection/firefox.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/firefox.html]: This note should be wrapped in a *warning* `o-media` object.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: Place the space outside the function call (`i18n("Step")`). Repeat everywhere.

File: html/captive-portal/templates/ssl_inspection/ios.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/ios.html]: Remove the parenthesis and wrap the note around a warning `o-media` object.

File: html/captive-portal/templates/ssl_inspection/opera.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/opera.html]: Wrap the note around a warning `o-media` object.

File: html/captive-portal/templates/ssl_inspection/opera.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/opera.html]: Remove the parenthesis and wrap the note around a warning `o-media` object.

File: html/captive-portal/templates/ssl_inspection/android.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/android.html]: Mac OSX => *macOS*

File: html/captive-portal/templates/ssl_inspection/osx.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/osx.html]: .. Mac-based browsers which use the operating system's built-in Keychain facility => .. *browsers which use macOS built-in Keychain facility*
Mac OSX => *macOS*

File: html/captive-portal/templates/ssl_inspection/osx.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/osx.html]: Remove the parenthesis and wrap the note around a warning `o-media` object.

File: html/captive-portal/templates/ssl_inspection/osx.html
Comment (cgx) [inline html/captive-portal/templates/ssl_inspection/osx.html]: Wrap the note around a warning `o-media` object.

**Review by cgx — CHANGES_REQUESTED** [review]

**Comment by extrafu** [comment]
@cgx Can you please make the changes before Thursday so we merge?

**Review by cgx — APPROVED** [review]

**Comment by cgx** [comment]
@julsemaan I'll let you merge once you approved the backend changes.

**Comment by julsemaan** [comment]
I'll dry the controller code a little bit and we'll be good to merge

**Comment by julsemaan** [comment]
The iOS instructions cannot work, they talk about a mobileconfig profile but the download link points to the same certificate file.

We'll need to have another link to a certificate enrollment profile for iOS and probably provide instructions and documentation on how to do that.

Or rework the backend to host the certificate on the server and then generate a mobileconfig file based on it

Nonetheless, this needs additional work to get iOS working fine according to the instructions

**Pull request merged**
